library/init/algebra/group.lean

```diff
@@ -55,8 +55,13 @@ meta def to_additive_attr : user_attribute (name_map name) name :=
     env ← get_env,
     dict ← to_additive_attr.get_cache,
     tgt ← to_additive_attr.get_param src,
-    (get_decl tgt >> skip) <|>
-      transport_with_dict dict src tgt }
+    let tgt' := match tgt, src with
+    | name.mk_string _ (name.mk_string _ _), _ := tgt
+    | _, name.mk_string _ pfx := tgt.update_prefix pfx
+    | _, _ := tgt
+    end,
```

Comment:
It would be nice if there was a version of `resolve_constant` that worked on names that don't exist, or a way to get the current namespace prefix at the location of the containing top-level command. There are still a lot of things that the `def` command does that are not accessible to tactics (parameters and variables too).

This commit also suggests the usage of `_root_` to disable namespace resolution. How feasible would it be to have `def` support this as well? People have asked before about how to declare things outside the current namespace, and this seems like a nice workaround.

Reply:
I undid this commit because it doesn't work without also adding the prefix management to the caching part, and I don't want to mix up parsing with the translation map. The best place to do the prefix management would be in a "before_set" handler that has time to preprocess the inputs before storing them in the definition, but this isn't available at the moment. I think I will add either a user command or a second user_attribute for handling this, but that part need not be done in the lean repo.